pkg/cmd/build/build.go

```diff
@@ -274,7 +274,7 @@ func extractFromContextAndDockerfile(context, dockerfile, svcName string) string
 		return dockerfile
 	}
 
-	if joinPath != dockerfile && filesystem.FileExistsAndNotDir(dockerfile) {
+	if joinPath != filepath.Join(".", dockerfile) && filesystem.FileExistsAndNotDir(dockerfile) {
```

Comment:
Could we sanitize the paths instead? this way to check equity seems very error prone